ios/brave-ios/Sources/BraveWallet/Crypto/Stores/UserAssetsStore.swift

```diff
@@ -275,6 +275,30 @@ public class UserAssetsStore: ObservableObject, WalletObserverStore {
     }
   }
 
+  @MainActor func checkDuplication(
+    address: String,
+    network: BraveWallet.NetworkInfo
+  ) async -> Bool {
+    let allUserAssetsExcludeDeleted = await assetManager.getAllUserAssetsInNetworkAssets(
+      networks: [network],
+      includingUserDeleted: false
+    )
+    let allTokens = await self.blockchainRegistry.allTokens(
+      in: [network],
+      includingUserDeleted: false
+    )
+    let existedInUserAsset =
+      allUserAssetsExcludeDeleted.flatMap(\.tokens).first(where: {
+        $0.contractAddress.caseInsensitiveCompare(address) == .orderedSame
+      }) != nil
+    let existedInTokenRegistry =
+      allTokens.flatMap(\.tokens).first(where: {
+        $0.contractAddress.caseInsensitiveCompare(address) == .orderedSame
+      }) != nil
```

Comment:
Is this going to cause issues with blockchain registry tokens that need a token id added?